src/AmalgamatorSession.ts

```diff
@@ -675,6 +696,35 @@ export class AmalgamatorSession extends LoggingDebugSession {
             ].customRequest('cdt-gdb-adapter/Memory', args);
             response.body = childResponse.body;
             this.sendResponse(response);
+        } else if (command === 'cdt-amalgamator/resumeAll') {
+            const [, threads] = await this.collectChildTheads();
+            await Promise.all(
+                threads.map(async (thread) => {
```

Comment:
This code is iterating through loops in two different ways, one using `map` and the other on line 708 using `for` with an index.

Also by doing a `map` you are having to surround the map with a `Promise.all` and have a dummy `Promise.resolve` at the end of it.

May I suggest for both loops you simply use [`for..of`] (https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html#forof-statements) loops

Reply:
I've updated it into 2 for loops

Reply:
Why didn't you use for of loops?

Reply:
Sorry, I updated using `for of` loops.